File: solutions/2026-05-03/solution.py
# Valid Palindrome
# Difficulty: Easy
# URL: https://leetcode.com/problems/valid-palindrome/

def canPermutePalindrome(s: str) -> bool:
  count = {}
  for char in s:
    count[char] = count.get(char, 0) + 1
  odd_count = 0
  for count_val in count.values():
    if count_val % 2 != 0:
      odd_count += 1
  return odd_count <= 1

File: solutions/2026-05-03/solution.java
// Valid Palindrome
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-palindrome/

public boolean canPermutePalindrome(String s) {
  int[] count = new int[128];
  for (char c : s.toCharArray()) {
    count[c]++;
  }
  int oddCount = 0;
  for (int countVal : count) {
    if (countVal % 2 != 0) {
      oddCount++;
    }
  }
  return oddCount <= 1;
}